### PR TITLE
Create game feature and Lobby Joining Implementation

### PR DESCRIPTION
**Pull request by zhasa18**
# Implement Game Lobby Creation and Join with Join Codes

## Summary of Changes

1. **Game Lobby Creation and Join Codes:**
   - **Lobby Creation:**
     - Implemented functionality for users to create game lobbies.
     - Each new lobby is assigned a unique join code from user input.
     - Added a `GamesController` with actions for creating and managing games.
   - **Joining Games:**
     - Users can join existing lobbies using a join code.
     - Added a modal on the landing page to enter join codes.
     - Implemented validations to handle invalid or duplicate join codes.

2. **Database Migrations and Associations:**
   - **Foreign Keys and Associations:**
     - Created migrations to establish foreign keys between the `User` and `Game` models through a join table `game_users`.
   - **Model Updates:**
     - Updated the `Game` and `User` models to reflect the new associations.
     - Added validations to the `Game` model for join code uniqueness and presence.

3. **Landing Page Updates:**
   - **List of Joined Games:**
     - Updated the landing page to display a list of games the logged-in user has joined.
     - Each game entry shows the game name.
     - Included button to navigate to the game lobby.

## Testing

- **Lobby Creation and Joining:**
  - Tested creating new game lobbies to ensure unique join codes are generated.
  - Verified that users can join lobbies using valid join codes.

- **Database Integrity:**
  - Ran migrations and checked that foreign keys are correctly set up.
  - Tested the associations by having users join and leave games.
  - Ensured that deleting a user or game updates related records without issues.

- **Landing Page Functionality:**
  - Tested navigation links to ensure they direct to the correct game lobbies.
  
-**Landing Page Image**
image

-**Game Creation Modal Image**
image

-**Join  Game Image**
image

-**Temporary Game Page Image**
image

### CODE REVIEW

**Comment by bjohnson66** [comment]
There are some missing gems from your gemfile

**Comment by zhasa18** [comment]
My bad, I think I resolved the issue.

File: Gemfile
Comment (halimyu) [inline Gemfile:1]: I’m s this crucial?

File: Rakefile
Comment (halimyu) [inline Rakefile:1]: Hmm

**Review by halimyu — APPROVED** [review]
Good work

**Review by bjohnson66 — COMMENTED** [review]
Works good, fixed gemfile, is proper branch this time

**Pull request merged**